third_party/move/move-model/bytecode/abstract_domain_derive/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Derive macro for `AbstractDomain`
+//!
+//! Currently we can only derive for structs.
+//! For tuple structs, the derived join pair-wise joins each field;
+//! for structs with named fields, the derived join pair-wise joins each field without #[no_join] attribute.
+
+use proc_macro::TokenStream;
+use quote::{quote, ToTokens};
+use syn::{self, parse_macro_input, DeriveInput, Fields};
+
+/// Given a field name, generates TokenStream of
+/// `join_result = JoinResult::combine(join_result, self.field_name.join(&other.field_name));`
```

Comment:
This comment oddly close to the code itself, although it uses `field_name` rather than the formal parameter `field`.  Perhaps better to have some text like "... generates a TokenStream to combine `join_result` with the `join` of `self.#field` with `&other.#field` (where `#field` is the literal value of the parameter `field`)."

(Although, as the parenthetical clause is true for all macros, it perhaps need not be included.)

Reply:
Perhaps it's easier to read the code, which is just one line?